design-docs/009-tigris-support.md

```diff
@@ -0,0 +1,50 @@
+# Tigris support
+
+[Tigris](https://www.tigrisdata.com/) is a globally distributed S3-compatible object store. Writes and reads get dispatched
+automatically to the geographically closest available region, and then the data gets automatically
+replicated to other regions.
+
+## Why is support needed
+
+Tigris doesn't offer read-after-create or read-after-update consistency by default. Data may be written to a region
+and read from another, where the data hasn't replicated yet.
+
+This means basic Icechunk guarantees cannot be hold. For example, something as simple as creating a repo can fails
+a high percentage of the time in certain geographic configurations.
+
+Tigris has built for us the ability to recover consistency by passing an `X-Tigris-Regions` header in the request. If
+writes and reads are done with the same region in the header, consistency is recovered. We also need `Cache-Control:no-cache`
+in the get requests. Of course, this comes at the price of performance, because reads no longer happens from the closest region anywhere in the world.
+
+An alternative would be to set the bucket as region-restricted. In that case all writes and reads go to the single region,
+but Icechunk cannot control the bucket configuration.
+
+## What we want
+
+Ideally most of the time users can use Icechunk in the most performant way, at least for reads. Even if it's
+slightly unsafe, users will want to make all reads from the closest regions.
+
+By default Icechunk should be perfectly safe and consistency, even at the price of slower performance.
+
+## How to achieve it
+
+* By default:
+  * When users create a Tigris `Storage` instance we force them to pass a region.
```

Comment:
This is already required for all S3-compatible object storage, no?

Reply:
Kind of, but not really. It's currently not enforced at the API level, but it's required by the S3-compatible `Storage`. It's in flow, we need to make a decision on how to improve. Enforcing mandatory argument feels weird, certain object stores don't need a region, on the other hand, not enforcing it for things like S3 is also wrong.